### PR TITLE
Add cloudbuild job to generate binaries and images

### PR DESCRIPTION
**Pull request by cpanato**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
- add cloudbuild job and mage rules to build the binaries and images and push those to our bom staging GCP project to be later promoted when needed 

1. I need to run a test before we merge this but it is pending some final bits in the GCP project
2. also need to create the postsubmit job which I will do soon

/assign @puerco @justaugustus 
cc @kubernetes-sigs/release-engineering 

#### Which issue(s) this PR fixes:

None

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: magefile.go [hunk 1]
```diff
@@ -114,6 +115,62 @@ func Build() error {
...
+	}
+
+	if err := sh.RunV("cd", "output"); err != nil {
+		return errors.Wrap(err, "cd into output directory")
+	}
+
+	if err := sh.RunV("./bom-linux-amd64", "output", "generate", "-f", "bom-darwin-amd64",
+		"-f", "bom-darwin-arm64", "-f", "bom-linux-amd64", "-f", "bom-linux-arm64",
+		"-f", "bom-windows-amd64.exe", "-d", "../", "-o", "bom-sbom.sdpx"); err != nil {
+		return errors.Wrap(err, "generating the bom")
```
Comment (puerco) [inline magefile.go:168]: This part is missing the images generated by ko before. But one thing to note is that to be able to add them we would need to alter the SBOM afterwards with the final registry.

The other option is that we wire SBOM generation into the promoter and attach the SBOMs with the signatures. I like the idea of having micro sboms in the images.

Comment (cpanato) [reply magefile.go:168]: sounds good, but this one is just for the binaries right? no images here

how do you want to proceed? we drop the bom generation from here and do at the promotion?

Comment (puerco) [reply magefile.go:168]: This one will create an SBOM with the binaries, source code and dependencies. Lets keep it here and work on attaching micro SBOMs to the images when promoted. I think I like that approach better :)

Comment (cpanato) [reply magefile.go]: i will remove this part for now and we can discuss that later, I would like to get the job running to we see that working :)

**Comment by cpanato** [comment]
run the job on cloudbuild, run example: https://console.cloud.google.com/cloud-build/builds/74cea60f-7a43-4b6d-96b7-3ba175c19101?project=533528353884

can find the test image: https://console.cloud.google.com/gcr/images/k8s-staging-bom/global/bom?project=k8s-staging-bom

and the binaries in https://console.cloud.google.com/storage/browser/k8s-staging-bom/add-cloudbuild?project=k8s-staging-bom&pageState=(%22StorageObjectListTable%22:(%22f%22:%22%255B%255D%22))&prefix=&forceOnObjectsSortingFiltering=false


I will create a post submit job to run the build when we merge to main or tag

sounds good @puerco ?

File: pkg/spdx/document_unit_test.go [hunk 2]
```diff
@@ -1,3 +1,19 @@
+/*
+Copyright 2021 The Kubernetes Authors.
```
Comment (saschagrunert) [inline pkg/spdx/document_unit_test.go:2]:
```suggestion
Copyright 2022 The Kubernetes Authors.
```

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Comment by cpanato** [comment]
Prow postsubmit job PR https://github.com/kubernetes/test-infra/pull/25436

**Comment by puerco** [comment]
Thank you @cpanato !
/lgtm

**Comment by cpanato** [comment]
ran here: https://console.cloud.google.com/cloud-build/builds;region=global/1b1847e1-148e-4c2c-a84f-080738c63db3?project=k8s-staging-bom

```
$ docker run gcr.io/k8s-staging-bom/bom:b68b231 version
  ____     ___    __  __
 | __ )   / _ \  |  \/  |
 |  _ \  | | | | | |\/| |
 | |_) | | |_| | | |  | |
 |____/   \___/  |_|  |_|
bom: A tool for working with SPDX manifests

GitVersion:    (devel)
GitCommit:     unknown
GitTreeState:  unknown
BuildDate:     unknown
GoVersion:     go1.17rc2
Compiler:      gc
Platform:      linux/arm64
```

checking why it did not get the git information like hash...

**Comment by cpanato** [comment]
the binary is fine, only the the image is missing the info

```
$ ./main_bom-darwin-arm64 version
  ____     ___    __  __
 | __ )   / _ \  |  \/  |
 |  _ \  | | | | | |\/| |
 | |_) | | |_| | | |  | |
 |____/   \___/  |_|  |_|
bom: A tool for working with SPDX manifests

GitVersion:    v0.2.2-26-gb68b231
GitCommit:     b68b231
GitTreeState:  clean
BuildDate:     2022-03-02T09:55:14Z
GoVersion:     go1.17rc2
Compiler:      gc
Platform:      darwin/arm64

```

**Pull request merged**